network/p2p/gossip/mempool.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package gossip
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+)
+
+var ErrDuplicateTx = errors.New("duplicate tx")
+
+type Mempool[T Gossipable] struct {
+	Txs     map[ids.ID]bool
```

Comment:
1. Do we want to expose this?
2. Same comment regarding naming consistency

Reply:
I apologize if my wording was confusing but I only explained that we could use a `map[ids.ID]struct{}` to clarify that it's equivalent to our existing `set.Set` implementation, which we should use instead.

Reply:
sounds good. updated.